src/Configurator/AddLinesConfigurator.php

```diff
@@ -0,0 +1,216 @@
+            }
+
+            if (!isset($patch['content'])) {
+                throw new \InvalidArgumentException(sprintf('The "content" key is required for the "add-lines" configurator for recipe "%s".', $recipe->getName()));
+            }
+            $content = $patch['content'];
+
+            $file = $this->path->concatenate([$this->options->get('root-dir'), $patch['file']]);
+            $warnIfMissing = isset($patch['warn_if_missing']) && $patch['warn_if_missing'];
+            if (!is_file($file)) {
+                $this->write([
+                    sprintf('Could not add lines to file <info>%s</info> as it does not exist. Missing lines:', $patch['file']),
+                    '<comment>"""</comment>',
+                    $content,
+                    '<comment>"""</comment>',
+                    '',
+                ], $warnIfMissing ? IOInterface::NORMAL : IOInterface::VERBOSE);
+
+                continue;
+            }
+
+            $this->write(sprintf('Patching file "%s"', $patch['file']));
+
+            if (!isset($patch['position'])) {
+                throw new \InvalidArgumentException(sprintf('The "position" key is required for the "add-lines" configurator for recipe "%s".', $recipe->getName()));
+            }
+            $position = $patch['position'];
+            if (!\in_array($position, self::VALID_POSITIONS, true)) {
+                throw new \InvalidArgumentException(sprintf('The "position" key must be one of "%s" for the "add-lines" configurator for recipe "%s".', implode('", "', self::VALID_POSITIONS), $recipe->getName()));
+            }
+
+            if (self::POSITION_AFTER_TARGET === $position && !isset($patch['target'])) {
+                throw new \InvalidArgumentException(sprintf('The "target" key is required when "position" is "%s" for the "add-lines" configurator for recipe "%s".', self::POSITION_AFTER_TARGET, $recipe->getName()));
+            }
+            $target = isset($patch['target']) ? $patch['target'] : null;
+
+            $this->patchFile($file, $content, $position, $target, $warnIfMissing);
+        }
+    }
+
+    public function unconfigure(Recipe $recipe, $config, Lock $lock): void
+    {
+        foreach ($config as $patch) {
+            if (!isset($patch['file'])) {
+                throw new \InvalidArgumentException(sprintf('The "file" key is required for the "add-lines" configurator for recipe "%s".', $recipe->getName()));
+            }
+
+            // Ignore "requires": the target packages may have just become uninstalled.
+            // Checking for a "content" match is enough.
+
+            $file = $this->path->concatenate([$this->options->get('root-dir'), $patch['file']]);
+            if (!is_file($file)) {
+                continue;
+            }
+
+            if (!isset($patch['content'])) {
+                throw new \InvalidArgumentException(sprintf('The "value" key is required for the "add-lines" configurator for recipe "%s".', $recipe->getName()));
```

Comment:
should we throw or should we ignore the invalid recipe ? throwing would break the whole composer command.

Reply:
All those validation rules should be implemented as part of the recipe validator for the CI of the recipes repo.

Reply:
Good catch! I've made a PR to add the linting rules - https://github.com/symfony-tools/recipes-checker/pull/14

I've also relaxed these to low-priority warnings (and skipping the rule). It will also help if we add more capabilities in the future to recipes, but they are perhaps on an older flex version (it won't work correctly, but no explosion).